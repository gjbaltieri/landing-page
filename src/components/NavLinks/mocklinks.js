export const links = [
  {
    children: 'google',
    link: 'http://google.com',
    newTab: false,
  },
  {
    children: 'twitter',
    link: 'http://twitter.com',
    newTab: false,
  },
  {
    children: 'uol',
    link: 'http://uol.com',
    newTab: false,
  },
];
