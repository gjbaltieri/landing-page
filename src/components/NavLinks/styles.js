import Styled, { css } from 'styled-components';
export const Container = Styled.nav`
${({ theme }) => css`
  display: flex;
  flex-flow: row wrap;
  @media ${theme.media.maxWidth768px} {
    flex-flow: column wrap;
    align-items: center;
  } ;
`}`;
