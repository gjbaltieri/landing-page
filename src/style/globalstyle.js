import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    h1, h2, h3, h4, h5, h6 {
        font-size: 62,5%;
        font-family: Montserrat, sans-serif;
    }

    * {
        margin: 0;
        padding: 0;
        box-sizing: border-box
    } 
`;
