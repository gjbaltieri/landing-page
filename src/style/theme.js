export const theme = {
  color: {
    background: {
      black: '#0D0D0D',
      white: '#F2F2F2',
    },
    heading: {
      black: '#333333',
      white: '#F2F2F2',
    },
  },
  font: {
    weight: '900',
    family: {
      default: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    size: {
      small: '0.8rem',
      base: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.2rem',
    },
  },
  media: {
    maxWidth768px: '(max-width: 768px)',
  },
  padding: {
    small: '0.8rem',
    base: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.2rem',
  },
};
