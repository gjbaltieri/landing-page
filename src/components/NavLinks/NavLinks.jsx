import P from 'prop-types';
import * as Styled from './styles';
import MenuLink from '../MenuLink/MenuLink.jsx';

export default function NavLinks({ links = [] }) {
  return (
    <Styled.Container>
      {links.map((link) => (
        <MenuLink key={link.link} {...link} />
      ))}
    </Styled.Container>
  );
}

NavLinks.prototype = {
  links: P.arrayOf(
    P.shape({
      children: P.string.isRequired,
      link: P.string.isRequired,
      newTab: P.bool,
    })
  ),
};
