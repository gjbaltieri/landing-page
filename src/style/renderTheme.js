import { render } from '@testing-library/react';
import { GlobalStyles } from './globalstyle';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';

export default function renderTheme(children) {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}
