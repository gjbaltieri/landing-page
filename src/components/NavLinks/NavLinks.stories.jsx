import React from 'react';
import NavLinks from './NavLinks';
import { links } from './mocklinks';

export default {
  title: 'NavLinks',
  component: NavLinks,
  args: { links: links },
  argTypes: {
    links: { type: '' },
  },
};

export const Template = (args) => (
  <div>
    <NavLinks {...args} />
  </div>
);
