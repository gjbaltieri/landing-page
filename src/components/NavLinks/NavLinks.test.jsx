/* eslint-disable testing-library/no-node-access */
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { links } from './mocklinks';
import { screen } from '@testing-library/react';
import renderTheme from '../../style/renderTheme';
import NavLinks from './NavLinks';
import { theme } from '../../style/theme';

describe('<NavLinks/>', () => {
  test('should render NavLinks with correct links length', () => {
    renderTheme(<NavLinks links={links} />);
    expect(screen.getAllByRole('link')).toHaveLength(links.length);
  });
  test('should render NavLinks with correct media query', () => {
    renderTheme(<NavLinks links={links} />);
    expect(screen.getByText(/google/i).parentElement).toHaveStyleRule(
      'flex-flow',
      'column wrap',
      { media: theme.media.maxWidth768px }
    );
  });
  test('should NavLinks not render links', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });
  test('should NavLinks match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={links} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
