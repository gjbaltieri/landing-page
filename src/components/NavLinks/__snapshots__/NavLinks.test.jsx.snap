// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks/> should NavLinks match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  color: #333333;
  font-size: 1.6rem;
  padding: 1.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
}

.c1:before {
  content: '';
  background: #54b3d6;
  display: block;
  position: absolute;
  bottom: -3px;
  left: 0;
  width: 0;
  height: 3px;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 0;
  width: 100%;
  height: 2px;
  background-color: white;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition: -webkit-transform 0.3s ease;
  -webkit-transition: transform 0.3s ease;
  transition: transform 0.3s ease;
}

.c1:hover::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<nav
  aria-label="Main menu"
  class="c0"
>
  <a
    class="c1"
    href="http://google.com"
    target="_self"
  >
    google
  </a>
  <a
    class="c1"
    href="http://twitter.com"
    target="_self"
  >
    twitter
  </a>
  <a
    class="c1"
    href="http://uol.com"
    target="_self"
  >
    uol
  </a>
</nav>
`;
